source/JsMaterialX/JsMaterialXFormat/JsXmlIo.cpp

```diff
@@ -13,25 +14,33 @@
 namespace ems = emscripten;
 namespace mx = MaterialX;
 
-extern "C"
+EMSCRIPTEN_BINDINGS(xmlio)
 {
-    EMSCRIPTEN_BINDINGS(xmlio)
-    {
-        ems::class_<mx::XmlReadOptions>("XmlReadOptions")
-            .constructor<>()
-            .property("readXIncludeFunction", &mx::XmlReadOptions::readXIncludeFunction)
-            .property("parentXIncludes", &mx::XmlReadOptions::parentXIncludes);
-        ems::class_<mx::XmlWriteOptions>("XmlWriteOptions")
-            .constructor<>()
-            .property("writeXIncludeEnable", &mx::XmlWriteOptions::writeXIncludeEnable)
-            .property("elementPredicate", &mx::XmlWriteOptions::elementPredicate);
+  ems::constant("MTLX_EXTENSION", mx::MTLX_EXTENSION);
+  ems::class_<mx::XmlReadOptions>("XmlReadOptions")
+      .constructor<>()
+      .property("readXIncludeFunction", &mx::XmlReadOptions::readXIncludeFunction)
+      .property("parentXIncludes", &mx::XmlReadOptions::parentXIncludes)
+      .property("readComments", &mx::XmlReadOptions::readComments)
+      .property("generateUniqueNames", &mx::XmlReadOptions::generateUniqueNames);
+  ems::class_<mx::XmlWriteOptions>("XmlWriteOptions")
+      .constructor<>()
+      .property("writeXIncludeEnable", &mx::XmlWriteOptions::writeXIncludeEnable)
+      .property("elementPredicate", &mx::XmlWriteOptions::elementPredicate);
+  ems::class_<mx::XmlExportOptions>("XmlExportOptions")
+      .constructor<>()
+      .property("mergeLooks", &mx::XmlExportOptions::mergeLooks)
+      .property("lookGroupToMerge", &mx::XmlExportOptions::lookGroupToMerge)
+      .property("flattenFilenames", &mx::XmlExportOptions::flattenFilenames)
+      .property("userDefinitionPath", &mx::XmlExportOptions::userDefinitionPath)
+      .property("userTexturePath", &mx::XmlExportOptions::userTexturePath)
+      .property("stringResolver", &mx::XmlExportOptions::stringResolver);
 
-        ems::function("readFromXmlString", ems::optional_override([](mx::DocumentPtr doc, std::string str, mx::XmlReadOptions readOptions = mx::XmlReadOptions()) {
-                     return mx::readFromXmlString(doc, (const std::string &)str, (const mx::XmlReadOptions *)&readOptions);
-                 }));
-
-        ems::function("writeToXmlString", ems::optional_override([](mx::DocumentPtr doc, mx::XmlWriteOptions writeOptions = mx::XmlWriteOptions()) {
-                     return mx::writeToXmlString(doc, (const mx::XmlWriteOptions *)&writeOptions);
-                 }));
-    }
+  BIND_FUNC_RAW_PTR("readFromXmlString", mx::readFromXmlString, 2, 3, mx::DocumentPtr , const std::string& , const mx::XmlReadOptions*);
+  BIND_FUNC_RAW_PTR("_readFromXmlFile", mx::readFromXmlFile, 3, 4, mx::DocumentPtr, mx::FilePath, mx::FileSearchPath, const mx::XmlReadOptions *);
```

Comment:
why the "_"? Function has 2 default params

Reply:
Good catch with the defaults. The _ is just to use it in the javascript wrapper since the actual api is defined in javascript. I could use the same name and rewrite it. I don't mind either